src/uri/authority.rs

```diff
@@ -103,15 +103,20 @@ impl Authority {
                 }
                 b':' => {
                     colon_cnt += 1;
-                },
+                }
                 b'[' => {
                     start_bracket = true;
+                    if has_percent {
+                        // Something other than the userinfo has a `%`, so reject it.
+                        return Err(ErrorKind::InvalidAuthority.into());
```

Comment:
Can you add a test case that verifies this error gets returned?

Reply:
Done.